### ISSUE TITLE
[REGRESSION]: Mouse class with parallel test leaks between tests in Firefox

### ISSUE THREAD

**Issue by Westbrook** [microsoft/playwright#27717]
**Context:**
- GOOD Playwright Version: 1.31.0
- BAD Playwright Version: I'm working on updating to 1.37.1, but it's not 100% clear where the issue started right now.
- Operating System: Linux and Mac, only contexts I've tested right now
- Extra: macOS M1 Max, locally, and Circle CI

**Code Snippet**

Help us help you! Put down a short code snippet that illustrates your bug and
that we can run and debug locally. For example:

Roughly as follows, but currently there's a lot of intermediary interaction that I've not fully worked through. I wanted to start the convo to see if anyone else had run into this as well.

```javascript
                    const page = session.browser.getPage(session.id);
                    for (const step of payload.steps) {
                        step.options = step.options || {};
                        if (step.position) {
                            await page.mouse[step.type](
                                Math.round(step.position[0]),
                                Math.round(step.position[1]),
                                step.options
                            );
                        } else {
                            await page.mouse[step.type as 'down' | 'up'](
                                step.options
                            );
                        }
                    }
```

**Describe the bug**
When testing in Firefox, and running tests in parallel, tests that leverage the mouse class seem to leak the mouse location between test contexts causing indeterminate results across a large test suite.

For my context, I'm leveraging `@web/test-runner` with Playwright as the browser launcher, much like is seen in https://github.com/modernweb-dev/web/issues/2374. Tests seem to intersect with each other and cause failures. Is there a setting, other than lowering concurrence to 1 that would allow for the functionality that was seen in 1.31.0 to be made possible in newer versions of the project?

**Comment by Westbrook** [comment]
Worth noting, the same test in Chromium and WebKit test in high parallelization with no issue.

**Comment by mxschmitt** [comment]
Can you provide us a minimal reproducible which works in Chromium and WebKit but fails in Firefox? This would help a ton to triage this issue further, thanks!

**Comment by Westbrook** [comment]
I’m working in how much can be taken out to trigger the issue. It’s time based in the parallelisation so the failures churn with each run and it’s also unclear, as yet, whether it’s parallelisation _generally_ or specific to how parallelisation works in `@web/test-runner` specifically that causes this situation.

**Comment by yury-s** [comment]
We need to be able to reproduce the problem to make this report actionable. Please file a new issue and link to this one if you manage to get a repro.
